examples/dna_language_models/dna_lm.ipynb

```diff
@@ -0,0 +1,642 @@
+    "    sequence = tokenizer(sequence)[\"input_ids\"]\n",
+    "    val_sequences.append(sequence)\n",
+    "    \n",
+    "\n",
+    "for i in range(0, len(ds['test'])):\n",
+    "    if dataset_limit and i == dataset_limit:\n",
+    "        break\n",
+    "    sequence = ds['test'][i]['sequence']\n",
+    "    sequence = \"candida_glabrata \" + \" \".join(get_kmers(sequence))\n",
+    "    sequence = tokenizer(sequence)[\"input_ids\"]\n",
+    "    test_sequences.append(sequence)\n",
+    "    \n",
+    "\n",
+    "train_labels = ds['train']['label']\n",
+    "test_labels = ds['test']['label']\n",
+    "val_labels = ds['validation']['label']\n",
+    "\n",
+    "if dataset_limit:\n",
+    "    train_labels = train_labels[0:dataset_limit]\n",
+    "    test_labels = test_labels[0:dataset_limit]\n",
+    "    val_labels = val_labels[0:dataset_limit]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0686955c-201a-427b-8bef-5c663edb85b8",
+   "metadata": {},
+   "source": [
+    "Finally, we create a Dataset object for each our sets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "445b4279-2446-46d6-af2a-ceb2638955c7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datasets import Dataset\n",
+    "\n",
+    "train_dataset = Dataset.from_dict({\"input_ids\": train_sequences, \"labels\": train_labels})\n",
+    "val_dataset = Dataset.from_dict({\"input_ids\": val_sequences, \"labels\": val_labels})\n",
+    "test_dataset = Dataset.from_dict({\"input_ids\": test_sequences, \"labels\": test_labels})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d05d51a7-b933-4793-95df-af7d4d510b13",
+   "metadata": {},
+   "source": [
+    "### 4. Train model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b5ce1985-c24e-4feb-a6d4-aacb909536f0",
+   "metadata": {},
+   "source": [
+    "Now, we'll train our DNA Language Model with the training dataset. We'll add a linear layer in the final layer of our language model, and then, train all the parameteres of our model with the training dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3a34b2c0-6205-4d48-b1a6-371b50ca42de",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import DataCollatorWithPadding\n",
+    "\n",
+    "data_collator = DataCollatorWithPadding(tokenizer=tokenizer)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "700540f4-0ab8-4f8a-a75c-416a6908af47",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "from torch import nn\n",
+    "\n",
+    "class DNA_LM(nn.Module):\n",
+    "    def __init__(self, model, num_labels):\n",
+    "        super(DNA_LM, self).__init__()\n",
+    "        self.model = model\n",
+    "        self.classifier = nn.Linear(768, num_labels)\n",
+    "        self.in_features = 768\n",
+    "        self.out_features = 2\n",
```

Comment:
Those are not really needed, are they?

Reply:
Apparently, it's needed else I get thrown an error when using Trainer that self.in_features and self.out_features is missing. I've made it a bit cleaner so the numbers aren't hard-coded.